modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1024,24 +1024,81 @@ public final HitInfo hitTest(Point2D point) {
         double x = point.getX() - getX();
         double y = point.getY() - getY() + getYRendering();
         GlyphList[] runs = getRuns();
-        int runIndex = 0;
+        int runIndex = findRunIndex(x, y, runs);
+
+        int textRunStart = 0;
+        int curRunStart = 0;
         if (runs.length != 0) {
+            textRunStart = findFirstRunStart(x, y, runs);
+            curRunStart = ((TextRun) runs[runIndex]).getStart();
+        }
+        TextLayout.Hit h = layout.getHitInfo((float)x, (float)y, getText(), textRunStart, curRunStart, false);
```

Comment:
I think `getText()` as a parameter is not needed here (it is passed at line 260 when it calls `setContent`).  Also, I think this should be `getTextInternal()` -- see the comment in that method.

I also think the `false` is not needed, see comments on the interface.

Reply:
Used `getTextInternal()` instead of `getText()`. I believe the text parameter is needed as mentioned before